core/engine/src/vm/opcode/args.rs

```diff
@@ -807,3 +939,654 @@ impl Argument for (VaryingOperand, ThinVec<u32>) {
         ((first.into(), rest), pos + 6 + total_len * 4)
     }
 }
+
+#[cfg(test)]
+mod test {
+
+  use std::vec;
+  use thin_vec::thin_vec;
+  use super::*;
+
+  fn test_write_arg<T: Argument>(test_data: T) -> Vec<u8> {
+    let mut v: Vec<u8> = Vec::new();
+    test_data.encode(&mut v);
+    return v
+  }
+
+  // This can fail depending on Argument/opcode combination given (too short)
+  fn test_read_arg<T: Argument>(opcode: Box<[u8]>, pos: usize) -> (T, usize) {
```

Comment:
If you always pass `0` to `pos` it's probably better to remove the pos argument.